analytics/analytics.go

```diff
@@ -95,11 +110,13 @@ func (dc *dataCollection) update() {
 
 	bs, ok := dc.node.Exchange.(*bitswap.Bitswap)
 	if !ok {
+		dc.reportHealthDueToError("failed to perform dc.node.Exchange.(*bitswap.Bitswap) type assertion")
```

Comment:
@zyyw Since we are storing this data in postgres database, should the return value be a long string? 意思是，我们需不需要把返回值简写成 "failed_BitSwap" 或者 "failed_PrivateKey"? Especially since in some functions, the err.Error() return value is being concatenated with the text string, the value could be a very long string. Thanks!

Reply:
i don't think we should worry about the error message being to long, which is not actually a long  err.error() message I believe

Reply:
> @zyyw Since we are storing this data in postgres database, should the return value be a long string? 意思是，我们需不需要把返回值简写成 "failed_BitSwap" 或者 "failed_PrivateKey"? Especially since in some functions, the err.Error() return value is being concatenated with the text string, the value could be a very long string. Thanks!

i don't think we should worry about the error message being to long, which is not actually a long err.error() message I believe

Reply:
Okay, sounds good, thx!